### PR TITLE
Added network_mode option to Docker container

### PR DESCRIPTION
**Pull request by bhean**
Hi,

First, sorry for my english...

I was _fighting_ some time with an issue in one of my roles, and finally it was related with the network of the Docker container created by Molecule.

I just added the `network_mode` parameter to the `create.yml` file, similar to `--net=host` in the `docker run` command.

According to the official documentation of the [Ansible Docker module](https://docs.ansible.com/ansible/latest/modules/docker_container_module.html), you can choose "_bridge_", "_host_", "_none_" or "_container:<name|id>_"

So now, you can set this value in the `platforms` node of the `molecule.yml` config file.

Thank you,

### CODE REVIEW

**Comment by retr0h** [comment]
Please update model, tests, and docs similar to #1413 for this new feature.  Thanks @bhean

**Comment by bhean** [comment]
You are right, I just made a fast change and commit, sorry....
Can yo check again?

File: molecule/driver/docker.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ class Docker(base.Base):
             networks:
               - name: foo
               - name: bar
+			network_mode: host
```
Comment (retr0h) [inline molecule/driver/docker.py]: looks like the indenting is off here.  Should be spaces.

```
 - name: bar
    network_mode: host
```

**Comment by retr0h** [comment]
Thanks @bhean, One in line comment.  Also, be sure to run `tox -e format` if you had not.

**Comment by bhean** [comment]
You are right, I used tabs instead spaces... now its fixed... (but `tox -e format` was working in both cases):

```
$ tox -e format
format create: /home/bhean/src/varios/molecule/.tox/format
format installdeps: -rrequirements.txt, -rtest-requirements.txt
format develop-inst: /home/bhean/src/varios/molecule
format installed: adal==1.0.2,ansible==2.6.3,ansible-lint==3.4.23,anyconfig==0.9.7,apache-libcloud==2.3.0,appdirs==1.4.3,applicationinsights==0.11.6,argcomplete==1.9.4,arrow==0.12.1,asn1crypto==0.24.0,attrs==18.2.0,azure-cli-core==2.0.35,azure-cli-nspkg==3.0.2,azure-common==1.1.11,azure-mgmt-batch==4.1.0,azure-mgmt-compute==2.1.0,azure-mgmt-containerinstance==0.4.0,azure-mgmt-containerregistry==2.0.0,azure-mgmt-containerservice==3.0.1,azure-mgmt-dns==1.2.0,azure-mgmt-keyvault==0.40.0,azure-mgmt-network==1.7.1,azure-mgmt-nspkg==2.0.0,azure-mgmt-rdbms==1.2.0,azure-mgmt-resource==1.2.2,azure-mgmt-sql==0.7.1,azure-mgmt-storage==1.5.0,azure-mgmt-web==0.32.0,azure-nspkg==2.0.0,azure-storage==0.35.1,Babel==2.6.0,bcrypt==3.1.4,binaryornot==0.4.4,boto==2.49.0,boto3==1.8.6,botocore==1.11.6,Cerberus==1.2,certifi==2018.8.24,cffi==1.11.5,chardet==3.0.4,click==6.7,click-completion==0.3.1,cliff==2.13.0,cmd2==0.9.4,colorama==0.3.9,cookiecutter==1.6.0,coverage==4.5.1,cryptography==2.3.1,debtcollector==1.20.0,decorator==4.3.0,deprecation==2.0.5,docutils==0.14,dogpile.cache==0.6.7,entrypoints==0.2.3,fasteners==0.14.1,flake8==3.5.0,future==0.16.0,git-url-parse==1.1.0,humanfriendly==4.16.1,idna==2.7,iso8601==0.1.12,isodate==0.6.0,jeepney==0.3.1,Jinja2==2.10,jinja2-time==0.2.0,jmespath==0.9.3,jsonpatch==1.23,jsonpointer==2.0,jsonschema==2.6.0,keyring==13.2.1,keystoneauth1==3.10.0,knack==0.3.3,MarkupSafe==1.0,mccabe==0.6.1,mock==2.0.0,-e git+git@github.com:bhean/molecule.git@184042dfb773c2e457fedb45a19a375e5b1ebad0#egg=molecule,monotonic==1.5,msgpack==0.5.6,msrest==0.4.29,msrestazure==0.4.31,munch==2.3.2,netaddr==0.7.19,netifaces==0.10.7,oauthlib==2.1.0,openstacksdk==0.17.2,os-client-config==1.31.2,os-service-types==1.3.0,osc-lib==1.11.1,oslo.config==6.4.0,oslo.i18n==3.21.0,oslo.serialization==2.27.0,oslo.utils==3.36.4,packaging==17.1,paramiko==2.4.1,pathspec==0.5.7,pbr==4.1.0,pexpect==4.6.0,pkginfo==1.4.2,pluggy==0.7.1,poyo==0.4.1,prettytable==0.7.2,psutil==5.4.6,ptyprocess==0.6.0,pur==5.0.0,py==1.6.0,pyasn1==0.4.4,pycodestyle==2.3.1,pycparser==2.18,pyflakes==1.6.0,Pygments==2.2.0,PyJWT==1.6.4,PyNaCl==1.2.1,pyOpenSSL==18.0.0,pyparsing==2.2.0,pyperclip==1.6.4,pytest==3.0.7,pytest-cov==2.5.1,pytest-helpers-namespace==2017.11.11,pytest-mock==1.10.0,pytest-verbose-parametrize==1.3.0,python-dateutil==2.7.3,python-gilt==1.2.1,python-ironicclient==2.5.0,python-keystoneclient==3.17.0,python-vagrant==0.5.15,pytz==2018.5,PyYAML==3.13,requests==2.19.1,requests-oauthlib==1.0.0,requests-toolbelt==0.8.0,requestsexceptions==1.4.0,rfc3986==1.1.0,s3transfer==0.1.13,SecretStorage==3.1.0,sh==1.12.14,shade==1.22.2,simplejson==3.16.0,six==1.11.0,stevedore==1.29.0,tabulate==0.8.2,testinfra==1.14.1,tox==3.2.1,-e git://github.com/AndreLouisCaron/tox-tags@0e681f527306279c7fbc94f68e95007d97de767e#egg=tox_tags,tqdm==4.25.0,tree-format==0.1.2,twine==1.11.0,urllib3==1.23,virtualenv==16.0.0,wcwidth==0.1.7,whichcraft==0.4.1,wrapt==1.10.11,yamllint==1.11.1,yapf==0.21.0
format runtests: PYTHONHASHSEED='570135711'
format runtests: commands[0] | yapf -i -r molecule/ test/
___________________________________________________________________________________________________________________________________ summary ____________________________________________________________________________________________________________________________________
  format: commands succeeded
  congratulations :)
```

**Comment by retr0h** [comment]
Thanks @bhean you rock \m/

**Pull request merged**